### PR TITLE
add test for repository.download_component_mapping

### PR DESCRIPTION
**Pull request by cklyyung**
Pull Request for issue #535.
Test cases:
1. Downloading 0 components
2. Downloading 2 components
3. Downloading 2 component with same e-tag as previous download

### CODE REVIEW

File: tests/test_repository.py [hunk 1]
```diff
@@ -237,6 +237,87 @@ def test_hg_log(fake_hg_repo):
...
+    responses.add(
+        responses.GET,
+        "https://index.taskcluster.net/v1/task/gecko.v2.mozilla-central.latest.source.source-bugzilla-info/artifacts/public/components.json",
+        status=200,
+        json={"README.txt": ["Core", "General"]},
+    )
+
+    repository.download_component_mapping()
+    assert len(repository.path_to_component) == 1
+    assert repository.path_to_component["README.txt"] == "Core::General"
```
Comment (marco-c) [inline tests/test_repository.py]: This case (1 component) is redundant with the previous one, so we can remove it.

File: tests/test_repository.py [hunk 2]
```diff
@@ -237,6 +237,87 @@ def test_hg_log(fake_hg_repo):
...
+        status=200,
+        headers={"ETag": "102"},
+    )
+
+    responses.add(
+        responses.GET,
+        "https://index.taskcluster.net/v1/task/gecko.v2.mozilla-central.latest.source.source-bugzilla-info/artifacts/public/components.json",
+        status=200,
+        json={"AUTHORS": ["mozilla.org", "Licensing"]},
+    )
```
Comment (marco-c) [inline tests/test_repository.py]: And of course when you remove that, you'll have to update this one to match Etag 101

Comment (cklyyung) [reply tests/test_repository.py]: Noted; I removed the test case of 1, and adjusted the e-tag.

File: tests/test_repository.py [hunk 3]
```diff
@@ -237,6 +237,69 @@ def test_hg_log(fake_hg_repo):
...
+
+    responses.reset()
+    responses.add(
+        responses.HEAD,
+        "https://index.taskcluster.net/v1/task/gecko.v2.mozilla-central.latest.source.source-bugzilla-info/artifacts/public/components.json",
+        status=200,
+        headers={"ETag": "101"},
+    )
+
+    responses.add(
```
Comment (marco-c) [inline tests/test_repository.py]: This one could be removed (the code will check the etag and if it's the same it won't download the file).

Comment (cklyyung) [reply tests/test_repository.py]: Ah, that's what my test case is, asserting that the response with the same etag won't be downloaded. If you think it's unnecessary, I will remove it when I have time later today.

Comment (marco-c) [reply tests/test_repository.py]: Exactly, so you just need to mock the HEAD request where the Etag is retrieved, this second request will never happen and so should not be mocked (this way we ensure it doesn't happen, which is what we want)

Comment (cklyyung) [reply tests/test_repository.py]: Ah, I understand, I misunderstood and thought you meant the whole test case.

**Pull request merged**